test/test_coldcard.py

```diff
@@ -0,0 +1,252 @@
+
+logging.info('Testing getkeypool: Not importable to privkey enabled wallet')
+keypool_desc = process_commands(dev_args + ['getkeypool', '--keypool', '0', '20'])
+import_result = wpk_rpc.importmulti(keypool_desc)
+assert(import_result[0]['success'] == False)
+
+logging.info('Testing getkeypool: Imports to non privkey enabled wallet keypool')
+import_result = wrpc.importmulti(keypool_desc)
+assert(import_result[0]['success'])
+for i in range(0, 21):
+    addr_info = wrpc.getaddressinfo(wrpc.getnewaddress())
+    assert(addr_info['hdkeypath'] == 'm/44\'/1\'/0\'/0/{}'.format(i))
+
+logging.info('Testing getkeypool: Imports to non privkey enabled wallet internal keypool')
+keypool_desc = process_commands(dev_args + ['getkeypool', '--keypool', '--internal', '0', '20'])
+import_result = wrpc.importmulti(keypool_desc)
+assert(import_result[0]['success'])
+for i in range(0, 21):
+    addr_info = wrpc.getaddressinfo(wrpc.getrawchangeaddress())
+    assert(addr_info['hdkeypath'] == 'm/44\'/1\'/0\'/1/{}'.format(i))
+
+logging.info('Testing getkeypool: --sh_wpkh uses correct derivation path')
+keypool_desc = process_commands(dev_args + ['getkeypool', '--keypool', '--sh_wpkh', '0', '20'])
+import_result = wrpc.importmulti(keypool_desc)
+assert(import_result[0]['success'])
+for i in range(0, 21):
+    addr_info = wrpc.getaddressinfo(wrpc.getnewaddress())
+    assert(addr_info['hdkeypath'] == 'm/49\'/1\'/0\'/0/{}'.format(i))
+keypool_desc = process_commands(dev_args + ['getkeypool', '--keypool', '--sh_wpkh', '--internal', '0', '20'])
+import_result = wrpc.importmulti(keypool_desc)
+assert(import_result[0]['success'])
+for i in range(0, 21):
+    addr_info = wrpc.getaddressinfo(wrpc.getrawchangeaddress())
+    assert(addr_info['hdkeypath'] == 'm/49\'/1\'/0\'/1/{}'.format(i))
+
+logging.info('Testing getkeypool: --wpkh uses correct derivation path')
+keypool_desc = process_commands(dev_args + ['getkeypool', '--keypool', '--wpkh', '0', '20'])
+import_result = wrpc.importmulti(keypool_desc)
+assert(import_result[0]['success'])
+for i in range(0, 21):
+    addr_info = wrpc.getaddressinfo(wrpc.getnewaddress())
+    assert(addr_info['hdkeypath'] == 'm/84\'/1\'/0\'/0/{}'.format(i))
+keypool_desc = process_commands(dev_args + ['getkeypool', '--keypool', '--wpkh', '--internal', '0', '20'])
+import_result = wrpc.importmulti(keypool_desc)
+assert(import_result[0]['success'])
+for i in range(0, 21):
+    addr_info = wrpc.getaddressinfo(wrpc.getrawchangeaddress())
+    assert(addr_info['hdkeypath'] == 'm/84\'/1\'/0\'/1/{}'.format(i))
+
+logging.info('Testing getkeypool: --account uses correct derivation path')
+keypool_desc = process_commands(dev_args + ['getkeypool', '--keypool', '--sh_wpkh', '--account', '3', '0', '20'])
+import_result = wrpc.importmulti(keypool_desc)
+assert(import_result[0]['success'])
+for i in range(0, 21):
+    addr_info = wrpc.getaddressinfo(wrpc.getnewaddress())
+    assert(addr_info['hdkeypath'] == 'm/49\'/1\'/3\'/0/{}'.format(i))
```

Comment:
just use `"` around it so you don't have to escape the hardening chars in this test file

Reply:
Done